src/Common/src/System/IO/PathInternal.Mono.cs

```diff
@@ -0,0 +1,259 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static partial class PathInternal
+    {
+        internal static readonly string ExtendedPathPrefix = PlatformHelper.IsWindows ? Windows.ExtendedPathPrefix : throw new PlatformNotSupportedException();
+
+        internal static readonly string UncPathPrefix = PlatformHelper.IsWindows ? Windows.UncPathPrefix : throw new PlatformNotSupportedException();
+
+        internal static readonly string UncExtendedPrefixToInsert = PlatformHelper.IsWindows ? Windows.UncExtendedPrefixToInsert : throw new PlatformNotSupportedException();
+
+        internal static readonly string UncExtendedPathPrefix = PlatformHelper.IsWindows ? Windows.UncExtendedPathPrefix : throw new PlatformNotSupportedException();
+
+        internal static readonly string DevicePathPrefix = PlatformHelper.IsWindows ? Windows.DevicePathPrefix : throw new PlatformNotSupportedException();
+
+        internal static readonly string ParentDirectoryPrefix = PlatformHelper.IsWindows ? Windows.ParentDirectoryPrefix : PlatformHelper.IsUnix ? Unix.ParentDirectoryPrefix : throw new PlatformNotSupportedException();
+
+        internal static readonly int MaxShortPath = PlatformHelper.IsWindows ? Windows.MaxShortPath : throw new PlatformNotSupportedException();
+
+        internal static readonly int MaxShortDirectoryPath = PlatformHelper.IsWindows ? Windows.MaxShortDirectoryPath : throw new PlatformNotSupportedException();
+
+        internal static readonly int MaxLongPath = PlatformHelper.IsWindows ? Windows.MaxLongPath : throw new PlatformNotSupportedException();
+
+        internal static readonly int DevicePrefixLength = PlatformHelper.IsWindows ? Windows.DevicePrefixLength : throw new PlatformNotSupportedException();
+
+        internal static readonly int UncPrefixLength = PlatformHelper.IsWindows ? Windows.UncPrefixLength : throw new PlatformNotSupportedException();
+
+        internal static readonly int UncExtendedPrefixLength = PlatformHelper.IsWindows ? Windows.UncExtendedPrefixLength : throw new PlatformNotSupportedException();
+
+        internal static readonly int MaxComponentLength = PlatformHelper.IsWindows ? Windows.MaxComponentLength : PlatformHelper.IsUnix ? Unix.MaxComponentLength : throw new PlatformNotSupportedException();
+
+        internal static char[] GetInvalidPathChars()
+        {
+            if (PlatformHelper.IsWindows)
```

Comment:
`return PlatformHelper.IsWindows ? Windows.GetInvalidPathChars() : Unix.GetInvalidPathChars();` is enough.

The redundant throw only guarantees that the method will never be inlined.